sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2540,7 +2540,7 @@ scatter_impl(AccessorTy acc, simd<T, N> vals, simd<uint32_t, N> offsets,
 
 template <typename T, int N, typename AccessorTy>
 ESIMD_INLINE ESIMD_NODEBUG std::enable_if_t<
-    (sizeof(T) <= 4) && (N == 1 || N == 8 || N == 16 || N == 32) &&
+    (sizeof(T) <= 4) && detail::isPowerOf2(N, 32) &&
```

Comment:
It is not equivalent of what the code "before" had. You add 2 and 4 here, but Ok, it seems you have E2E tests verifying it works correctly.

Additionally, please move this condition to the body of the function and make it static_assert() with meaningful messages. static_assert fail is much easier for understanding than an endless list of notes listing a candidates that do not match due to various reasons.

Reply:
Anyway, thank you for turning SFINAE condition to static_assert. 

If allow N=2, and N=4, then allow testing of it even without #ifdef __ESIMD_GATHER_SCATTER_LLVM_IR.